src/server/collections/videos-db.js

```diff
@@ -111,9 +117,62 @@ function getAllVideo(callback) {
   });
 }
 
+function insertVideoToDatabase(userInfo, videoInfos, token, callback) {
+  MongoClient.connect(url, (err, db) => {
+    if (err) {
+      console.log(err.name + ':' + err.message);
+      return callback(undefined);
+    }
+    let videosDB = db.collection('videos');
+    let uploadVideoInfos = {
+      'videoUrl': videoInfos.videoUrl,
+      'videoId': null,
+      'videoDetails': {
+        'title': videoInfos.videoTitle,
+        'views': 0,
+        'LikeStatus': [],
+        'publishDate': Date.now(),
+        'time': '5:00',
+        'preview': videoInfos.preview,
+      },
+      'uploader': {
+        'name': userInfo.username,
+        'userId': userInfo._id,
+        'avatar': userInfo.avatar,
+        'subscribers': userInfo.subscribers,
+      },
+      'commentInfos': [],
+    };
+
+    videosDB.insertOne(uploadVideoInfos, (err, result) => {
+      if (err) {
+        return callback(undefined);
+      }
+      videosDB.update(
```

Comment:
if you set the videoId during the response generation you dont need this update anymore